data/tests/test_emission_calculations.py

```diff
@@ -26,7 +26,7 @@ def test_get_n_prep_data_from_smhi(self):
         df_input = pd.DataFrame(pd.read_excel(path_input_df))
         df_result = get_n_prep_data_from_smhi(df_input)
         result_columns = df_result.columns.to_list()[4:] # Skip the first 4 columns which is the 'index'
-        expected_columns = [1990, 2000, 2005, 2010, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022]
+        expected_columns = [1990, 2000, 2005, 2010, 2015, 2016, 2017, 2018, 2019, 2020, 2021]
```

Comment:
Do I understand the following correctly?

The main goal of changes to the tests is to avoid having to update tests every year to reflect the latest year with data. This was made possible by testing explicit years (by adding 2021 in multiple calls below), rather than falling back to the implicit behavior of `LAST_YEAR_WITH_SMHI_DATA`.

Is it safe to assume that we don't need to test calculations for the latest year **unless the data format changes** (like the input excel file)?

---

If my interpretation is correct, this looks good.

However, I can't get the local tests to pass, (maybe due to the NaN value for Kumla)

Reply:
Exactly, the idea is to keep the values that have been verified as correct test results without having to update them every year (thus avoiding a whole lot of hassle).

And yes, we can assume that if the data remain in the same format (and even in several scenarios where the format changes), the tests can still look the same (with the exception of the first one) since they are testing mathematical calculations.